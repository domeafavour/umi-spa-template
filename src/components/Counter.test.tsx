import { act, render, renderHook } from '@testing-library/react';
import { Counter, useCounter } from './Counter';

describe('useCounter', () => {
  it('should render without crash', () => {
    const { result } = renderHook(() => useCounter());
    expect(result.current[0]).toBe(0);
    act(() => {
      result.current[1].increment();
    });
    expect(result.current[0]).toBe(1);
  });
});

describe('Counter', () => {
  it('should render without crash', () => {
    const { container, getByRole, rerender } = render(<Counter />);
    expect(container).toMatchSnapshot();
    act(() => {
      getByRole('button').click();
    });
    rerender(<Counter />);
    expect(getByRole('button').textContent).toBe('1');
  });
});
