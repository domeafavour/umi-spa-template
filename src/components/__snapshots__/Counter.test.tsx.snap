// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render without crash 1`] = `
<div>
  <button
    type="button"
  >
    0
  </button>
</div>
`;
